jbi/services/jira.py

```diff
@@ -253,33 +261,49 @@ def create_jira_issue(
     ):
         """Create a Jira issue with basic fields in the project and return its key."""
         bug = context.bug
-        logger.debug(
-            "Create new Jira issue for Bug %s",
-            bug.id,
-            extra=context.model_dump(),
-        )
         fields: dict[str, Any] = {
             "summary": bug.summary,
             "issuetype": {"name": issue_type},
             "description": description[:JIRA_DESCRIPTION_CHAR_LIMIT],
             "project": {"key": context.jira.project},
         }
-
-        jira_response_create = self.client.create_issue(fields=fields)
+        logger.debug(
+            "Creating new Jira issue for Bug %s",
+            bug.id,
+            extra={"fields": fields, **context.model_dump()},
+        )
+        try:
+            response = self.client.create_issue(fields=fields)
+        except requests.HTTPError as exc:
+            assert exc.response is not None
+            try:
+                response_json = exc.response.json()
+                logger.exception(
+                    "Failed to create issue for Bug %s",
+                    bug.id,
+                    extra={"response": response_json, **context.model_dump()},
+                )
+                message = ", ".join(str(exc).split("\n"))
+                raise JiraCreateError(message) from exc
+            except json.JSONDecodeError:
+                # If the response is not JSON, use the response text
+                logger.exception(
+                    "Failed to create issue for Bug %s",
+                    bug.id,
+                    extra={"response": exc.response.text, **context.model_dump()},
+                )
+                raise JiraCreateError(str(exc)) from exc
```

Comment:
Wouldn't this be more concise? Is the `join`/`split` specific to the http exception?

```suggestion
            try:
                response = exc.response.json()
            except json.JSONDecodeError:
                # If the response is not JSON, use the response text
                response = exc.response.text
            
            logger.exception(
                "Failed to create issue for Bug %s",
                bug.id,
                extra={"response": response, **context.model_dump()},
            )
            message = ", ".join(str(exc).split("\n"))
            raise JiraCreateError(message) from exc
```

Reply:
Good call. I changed things up a big more in [e7aa369](https://github.com/mozilla/jira-bugzilla-integration/pull/767/commits/e7aa36959f50b8d17602e9fa42f0c1e3759bee72) -- lmk what you think